crates/cli/src/main.rs

```diff
@@ -158,29 +253,126 @@ fn output_result_debug<R: std::fmt::Debug>(result: R) {
 /// # Arguments
 /// * `result` - A serde serializable result.
 #[cfg(feature = "json")]
-fn output_result_json<R: serde::Serialize>(result: R) {
-    serde_json::to_writer_pretty(std::io::stdout(), &result).unwrap();
+fn output_result_json<T: serde::Serialize>(result: T) {
+    println!("{}", serde_json::to_string(&result).unwrap());
```

Comment:
nitpick: It is slightly more memory-efficient to use to_writer for outputting to stdout (or files).

Reply:
yeah ik this, its like this due to the writer not having a newline on stdout

Reply:
Ah, my bad that makes sense.